Attempts to make it available on st3.

glymehrvrd opened the pull request:
Hi, I have done some coding to make it available on st3.
works well on Ubuntu, test function is not tested.

glymehrvrd committed:
Summary: modified to adapt st3. only completion is done.
- remove class SublimerlGlobal, extract its methods into outter
  functions.
+ add sublimerl.py, make it the entry of this plugin.
* move original *.py into package sublimerl_lib.

glymehrvrd committed:
Summary:
+ add some comments.
* modify completion templete to make the last jumping point exactly right to the rightest braket.

glymehrvrd committed:
Summary: make all functions available on st3.
- man function, I use Zeal instead.
+ erlang parser can detect -compile([export_all]) now.
+ add function search, format code into context menu.
* if rebar.config and ebin cannot be found, use editing file's path as
root.

ostinelli commented:
I see that there are a lot of changes that are related to code style preferences (for example, [here](https://github.com/glymehrvrd/SublimErl/commit/2797d484d4c57af998a8e1c19a7876b048b6b59f#diff-fc2671a5116df83185b1e77775baa559R30)), and to the fact that different spacing is used.
Also, there are a lot of different commits.

Please adapt to code styling, provide only real changes, and stash everything into a single commit if you want me to review.

Best,
r.

ostinelli commented:
Also, there already is work done for ST3 here: #78. Why wasn't this good and you provided an alternative?

glymehrvrd commented:
The code #78 supports windows only, and I found the code #78 still uses SublimErlGlobal. I think using SublimErlGlobal is not a good idea since st3 loads plugins async. So I do lots of changes to make the code more adapt to st3.
I will organize my commits later when I have free time :)

glymehrvrd committed:
Summary: Translate all tabs to four spaces.

glymehrvrd committed:
Summary: Regenerate Erlang-Libs.sublime-completions

glymehrvrd committed:
Summary: add *.bak to .gitignore

glymehrvrd commented:
@ostinelli , hi, I have tried to make my commits more tidy, see is that acceptable for u. :D

glymehrvrd committed:
Summary: fix a bug in sublimerl_libparser.py

glymehrvrd committed:
Summary: another bug in sublimerl_libparser.py is fixed

glymehrvrd committed:
Summary: Regenerate sublime-completions.

glymehrvrd committed:
Summary: remove files under completion folder.

glymehrvrd committed:
Summary:
- remove redundant code.
* fix whether context is enabled.

Commit:
Fix erl_scan:token_info deprecated warning

Warning: erl_scan:token_info/2: deprecated (will be removed in OTP 19); use erl_scan:{category,column,line,location,symbol,text}/1 instead

glymehrvrd committed:
Merge pull request #1 from d-kurochkin/master

Fix erl_scan:token_info deprecated warning
